modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIDenyPolicyTestCase.java

```diff
@@ -0,0 +1,362 @@
+        String apiVersion = "1.0";
+        String url = getGatewayURLHttp() + "jaxrs_basic/services/customers/customerservice";
+        APIRequest apiRequest = new APIRequest(apiName, apiContext, new URL(url));
+        apiRequest.setVersion(apiVersion);
+        apiRequest.setTiersCollection(APIMIntegrationConstants.API_TIER.UNLIMITED);
+        apiRequest.setTier(APIMIntegrationConstants.API_TIER.UNLIMITED);
+        apiRequest.setProvider(user.getUserName());
+
+        //Add the API using the API publisher.
+        HttpResponse postResponse = restAPIPublisher.addAPI(apiRequest);
+        String apiId = postResponse.getData();
+        waitForAPIDeployment();
+        //create Deny Policy
+        boolean conditionStatus = true;
+        Object conditionValue = "/deny/1.0";
+
+        blockingConditionDTO = new BlockingConditionDTO();
+        blockingConditionDTO.setConditionStatus(conditionStatus);
+        blockingConditionDTO.setConditionValue(conditionValue);
+        blockingConditionDTO.setConditionType(BlockingConditionDTO.ConditionTypeEnum.API);
+
+        ApiResponse<BlockingConditionDTO> addedPolicy = restAPIAdmin.addDenyThrottlingPolicy(blockingConditionDTO);
+
+        Assert.assertEquals(addedPolicy.getStatusCode(), HttpStatus.SC_CREATED);
+        BlockingConditionDTO addedBlockingConditionDTO = addedPolicy.getData();
+        String policyId = addedBlockingConditionDTO.getConditionId();
+        Assert.assertNotNull(policyId, "The deny policy ID cannot be null or empty");
+        blockingConditionDTO.setConditionId(policyId);
+    }
+
+    @Test(groups = {"wso2.am"}, description = "Test get API deny policy", dependsOnMethods = "testAddAPIDenyPolicy")
+    public void testGetAddedDenyPolicy() throws ApiException {
+        ApiResponse<BlockingConditionDTO> response = restAPIAdmin.getDenyThrottlingPolicy(blockingConditionDTO.getConditionId());
+        String retrievedConditionId = response.getData().getConditionId();
+
+        Assert.assertEquals(blockingConditionDTO.getConditionId(), retrievedConditionId);
+    }
+
+    @Test(groups = {"wso2.am"}, description = "Test add API deny policy with non existing context", dependsOnMethods = "testGetAddedDenyPolicy")
+    public void testAddDenyPolicyWithNonExistingContext() {
+
+        //create Deny Policy
+        boolean conditionStatus = true;
+        Object conditionValue = "/denyNonExisting/1.0.0/";
+
+        BlockingConditionDTO blockingConditionDTO = new BlockingConditionDTO();
+        blockingConditionDTO.setConditionStatus(conditionStatus);
+        blockingConditionDTO.setConditionValue(conditionValue);
+        blockingConditionDTO.setConditionType(BlockingConditionDTO.ConditionTypeEnum.API);
+        try {
+            ApiResponse<BlockingConditionDTO> addedPolicy = restAPIAdmin.addDenyThrottlingPolicy(blockingConditionDTO);
+        } catch (ApiException e) {
+            Assert.assertEquals(e.getCode(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
```

Comment:
Does this give a 500? Ideally it should give 400.. 
If 500, please create an issue, and mention about this test case as well. So if someone fixing the issue, he can fix the test as well.

Reply:
Yes It gives 500 not 400. Will create an issue.

Reply:
https://github.com/wso2/product-apim/issues/11213